src/OrchardCore/Orchard.Recipes.Implementations/Services/RecipeExecutor.cs

```diff
@@ -259,7 +259,7 @@ private void EvaluateJsonTree(IScriptingManager scriptingManager, JToken node)
                     // Evaluate the expression while the result is another expression
                     while (value.StartsWith("[") && value.EndsWith("]"))
                     {
-                        value = value.Trim('[', ']');
+                        value = value.Substring(1, value.Length - 2);
```

Comment:
@sebastienros I replaced Trim because it caused a bug on expressions like this: `"[js:parameters('extraSettings.Code')[0]]"` because it removes last two brackets instead of only ones causing an exception later when the expression is parsed.